pkg/cli/project/delete.go

```diff
@@ -0,0 +1,65 @@
+package project
+
+import (
+	"context"
+	"fmt"
+
+	projectc "github.com/fuseml/fuseml-core/gen/http/project/client"
+	"github.com/fuseml/fuseml-core/pkg/cli/client"
+	"github.com/fuseml/fuseml-core/pkg/cli/common"
+	"github.com/spf13/cobra"
+)
+
+// DeleteOptions holds the options for 'project delete' sub command
+type DeleteOptions struct {
+	client.Clients
+	global *common.GlobalOptions
+	Name   string
+}
+
+// NewDeleteOptions creates a ProjectDeleteOptions struct
+func NewDeleteOptions(o *common.GlobalOptions) *DeleteOptions {
+	return &DeleteOptions{global: o}
+}
+
+// NewSubCmdProjectDelete creates and returns the cobra command for the `project delete` CLI command
+func NewSubCmdProjectDelete(gOpt *common.GlobalOptions) *cobra.Command {
+
+	o := NewDeleteOptions(gOpt)
+
+	cmd := &cobra.Command{
+		Use:   `delete {-n|--name NAME}`,
+		Short: "Delete projects.",
+		Long:  `Delete a project from FuseML`,
```

Comment:
I think it would be nice to say that deleting a project also deletes all codesets associated with the project, or it cannot be deleted if there are existing codesets associated with it?

Reply:
Actually, I just remembered that I did not implement delete operation yet.... 

But I think i'd rather not allow deleting project when codesets still exist